examples/ruby_example/spec/ruby_example_spec.rb

```diff
@@ -88,4 +96,104 @@ def initialize(a, b)
       expect(e.class).to eq CcacheRedisError
     end
   end
+
+  describe "GC" do
+    def insert
+      ruby_store.insert("key", Foo.new(1, 2))
+    end
+
+    it "inserted value should not be garbage collected" do
+      insert
+
+      GC.start # trigger gc manually
+
+      val = ruby_store.get("key")
+
+      expect(val.class).to eq Foo
+      expect(val.a).to eq 1
+      expect(val.b).to eq 2
+    end
+  end
+
+  describe "GC example" do
+    def memory_usage
+      _, size = `ps ax -o pid,rss | grep -E "^[[:space:]]*#{$$}"`.strip.split.map(&:to_i)
+
+      size
+    end
+
+    def insert
+      ruby_store.insert("key", Foo.new(1, 2))
+    end
+
+    def insert_random(ruby_store)
+      ruby_store.insert("key", Foo.new(rand, rand))
+    end
+
+    it 'inserted value should not be garbage collected' do
+      insert
+
+      GC.start # trigger gc manually
+
+      val = ruby_store.get("key")
+
+      expect(val.class).to eq Foo
+      expect(val.a).to eq 1
+      expect(val.b).to eq 2
+    end
+
+    it 'Ruby should collect unreferenced objects' do
+      skip "when ruby_store has been reclaimed, it should clear Rust struct"
+
+      total = 100_000
+      ruby_store
+
+      before = ObjectSpace.count_objects[:TOTAL]
+      memory_before = memory_usage
+      ruby_store = RubyStore.new("redis://127.0.0.1/")
+
+      total.times { insert_random(ruby_store) }
+
+      ruby_store = nil
+      sleep 0.1
+
+      GC.start
```

Comment:
when ruby_store has been cleared, Rust structs should be destroyed too. it does not work here. Will consider this in the future.